synapse/app/generic_worker.py

```diff
@@ -187,9 +189,8 @@ async def on_POST(self, request: Request, device_id: Optional[str]):
                 # If the header exists, add to the comma-separated list of the first
                 # instance of the header. Otherwise, generate a new header.
                 if x_forwarded_for:
-                    x_forwarded_for = [
-                        x_forwarded_for[0] + b", " + previous_host
-                    ] + x_forwarded_for[1:]
+                    x_forwarded_for = [x_forwarded_for[0] + b", " + previous_host]
+                    x_forwarded_for.extend(x_forwarded_for[1:])
```

Comment:
`requestHeaders.getRawHeaders` is defined as returning a `Sequence` so mypy complained about this. This should be identical, in reality it *does* return a `List`.

Reply:
I wouldn't have spotted an error on the LHS! I was curious, and reproducing it gave me 

> `Unsupported operand types for + ("List[bytes]" and "Sequence[bytes]")

which I didn't realise was a thing. But it is:

```python
>>> [1] + (2,3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate list (not "tuple") to list
```

Good to know!

Reply:
I would have thought the operator would just care for the RHS to be an iterable, but I guess not. 🤷